src/app/cli/src/commands/pull_command.rs

```diff
@@ -115,19 +115,26 @@ impl PullCommand {
         listener: Option<Arc<dyn PullMultiListener>>,
         current_account_name: &odf::AccountName,
     ) -> Result<Vec<PullResponse>, CLIError> {
-        let dataset_any_refs: Vec<_> = if !self.all {
-            filter_datasets_by_any_pattern(
-                self.dataset_registry.as_ref(),
-                self.search_svc.clone(),
-                self.refs.clone(),
-                current_account_name,
-                self.tenancy_config,
+        let dataset_any_refs = DatabaseTransactionRunner::new(self.catalog.clone())
+            .transactional_with2(
+                |dataset_registry: Arc<dyn DatasetRegistry>,
+                 search_svc: Arc<dyn SearchServiceRemote>| async move {
+                    if !self.all {
+                        filter_datasets_by_any_pattern(
+                            dataset_registry.as_ref(),
+                            search_svc,
+                            self.refs.clone(),
+                            current_account_name,
+                            self.tenancy_config,
+                        )
+                        .try_collect()
+                        .await
+                    } else {
+                        Ok(vec![])
+                    }
+                },
```

Comment:
If you move this block to a separate method, you could utilize #[transactional_method2] macro to make it more compact.